cmd/archeio/main_test.go

```diff
@@ -0,0 +1,135 @@
+	for _, sc := range s.scenarios {
+		server := httptest.NewServer(s.handler)
+		defer server.Close()
+		client := server.Client()
+		if sc.request.redirect == false {
+			client.CheckRedirect = func(req *http.Request, via []*http.Request) error {
+				return http.ErrUseLastResponse
+			}
+		}
+		resp, err := client.Get(server.URL + sc.request.path)
+		if err != nil {
+			t.Errorf("Error requesting fake backend, %v", err)
+		}
+		for _, test := range s.tests {
+			test(resp, sc)
+		}
+	}
+}
+
+func defaultTestFuncs(t *testing.T) []func(resp *http.Response, sc scenario) {
+	return []func(resp *http.Response, sc scenario){
+		func(resp *http.Response, sc scenario) {
+			if resp.StatusCode != sc.expected.statusCode {
+				t.Errorf("Expected status code '%v' but received '%v', scenario: %#v, resp: %#v", resp.StatusCode, sc.expected.statusCode, sc, resp)
+			}
+		},
+		func(resp *http.Response, sc scenario) {
+			if sc.expected.path != "" && resp.Request.URL.Path != sc.expected.path {
+				t.Errorf("Expected path '%v' but received '%v', scenario: %#v, resp: %#v", resp.Request.URL.Path, sc.expected.url, sc, resp)
+			}
+			if sc.expected.url != "" && defaultUpstreamRegistry+resp.Request.URL.Path != sc.expected.url {
+				t.Errorf("Expected url '%v' but received '%v', scenario: %#v, resp: %#v", defaultUpstreamRegistry+resp.Request.URL.Path, sc.expected.url, sc, resp)
+			}
+		},
+	}
+}
+
+func TestMakeHandler(t *testing.T) {
+	suite := &suite{
+		handler: makeHandler(defaultUpstreamRegistry),
+		scenarios: []scenario{
+			{
+				request:  request{path: "/", redirect: false},
+				expected: expected{path: "/", statusCode: http.StatusNotFound},
+			},
+			// when not redirecting
+			{
+				request:  request{path: "/v2/", redirect: false},
+				expected: expected{url: defaultUpstreamRegistry + "/v2/", statusCode: http.StatusPermanentRedirect},
+			},
+			{
+				request:  request{path: "/v1/", redirect: false},
+				expected: expected{url: defaultUpstreamRegistry + "/v1/", statusCode: http.StatusPermanentRedirect},
+			},
+			// when redirecting, results from k8s.gcr.io
+			{
+				request:  request{path: "/v2/", redirect: true},
+				expected: expected{url: defaultUpstreamRegistry + "/v2/", statusCode: http.StatusUnauthorized},
+			},
+			{
+				request:  request{path: "/v1/", redirect: true},
+				expected: expected{url: defaultUpstreamRegistry + "/v1/", statusCode: http.StatusNotFound},
+			},
+		},
+		tests: defaultTestFuncs(t),
+	}
+	suite.runTestSuite(t)
+}
+
+func TestDoV2(t *testing.T) {
+	suite := &suite{
+		handler: http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			doV2(w, r, defaultUpstreamRegistry)
+		}),
+		scenarios: []scenario{
+			{
+				request:  request{path: "/v2/", redirect: false},
+				expected: expected{url: defaultUpstreamRegistry + "/v2/", statusCode: http.StatusPermanentRedirect},
+			},
+		},
+		tests: defaultTestFuncs(t),
+	}
+	suite.runTestSuite(t)
+}
+
+func TestDoV1(t *testing.T) {
```

Comment:
@ameukam mentioned we should drop v1 support in another PR but I didn't get that follow-up filed, do you want to do that and then drop it here? https://github.com/kubernetes-sigs/oci-proxy/pull/28#discussion_r831573854

I'll file an issue to track either way, and check back and do it if not in the morning my time 🙃

Reply:
filed https://github.com/kubernetes-sigs/oci-proxy/issues/34 to track